subworkflows/local/profiling.nf

```diff
@@ -346,9 +346,8 @@ workflow PROFILING {
                                 .multiMap {
                                     it ->
                                         reads: [it[0] + it[2], it[1]]
-                                        db: it[3]
+                                        db: [ it[2], it[3] ]
                                 }
-
         // diamond only accepts single output file specification, therefore
         // this will replace output file!
         ch_diamond_reads_format = params.diamond_save_reads ? 'sam' : params.diamond_output_format
```

Comment:
@jfy133  Do you know why we got this error when running `test_nonprocessing.config`? I have checked the `ch_input_for_diamond.reads` for `test.config` and `test_nonprocessing.config` and I can't see any difference between them except the reads of multiple runs were merged in the `test.config`.
```
[DUMP: diamond_test_reads] [['id':'2611', 'run_accession':'ERR5766174', 'instrument_platform':'ILLUMINA', 'single_end':true, 'is_fasta':true, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fasta/ERX5474930_ERR5766174_1.fa.gz]]
[DUMP: diamond_test_reads] [['id':'ERR3201952', 'instrument_platform':'OXFORD_NANOPORE', 'single_end':true, 'is_fasta':false, 'type':'long', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/home/lili.li/taxprofiler_update_tools_new/work/9f/ca4c96cee207ebba5e6d645ba7de20/ERR3201952_ERR3201952.unmapped_other.fastq.gz]]
[DUMP: diamond_test_reads] [['id':'2613', 'instrument_platform':'ILLUMINA', 'single_end':true, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/home/lili.li/taxprofiler_update_tools_new/work/f9/0c53ee2b2bccb1d4e1866a755962c8/2613_ERR5766181.unmapped.fastq.gz]]
[DUMP: diamond_test_reads] [['id':'2612', 'instrument_platform':'ILLUMINA', 'single_end':true, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/home/lili.li/taxprofiler_update_tools_new/work/b2/ed806ae5d462922f2c183229e9fce0/2612.merged.fastq.gz]]


[DUMP: diamond_test-noprocessing_reads] [['id':'2611', 'run_accession':'ERR5766174', 'instrument_platform':'ILLUMINA', 'single_end':true, 'is_fasta':true, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fasta/ERX5474930_ERR5766174_1.fa.gz]]
[DUMP: diamond_test-noprocessing_reads] [['id':'2612', 'run_accession':'ERR5766176', 'instrument_platform':'ILLUMINA', 'single_end':false, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fastq/ERX5474932_ERR5766176_1.fastq.gz, /nf-core/test-datasets/taxprofiler/data/fastq/ERX5474932_ERR5766176_2.fastq.gz]]
[DUMP: diamond_test-noprocessing_reads] [['id':'2612', 'run_accession':'ERR5766176_B', 'instrument_platform':'ILLUMINA', 'single_end':false, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fastq/ERX5474932_ERR5766176_B_1.fastq.gz, /nf-core/test-datasets/taxprofiler/data/fastq/ERX5474932_ERR5766176_B_2.fastq.gz]]
[DUMP: diamond_test-noprocessing_reads] [['id':'2612', 'run_accession':'ERR5766180', 'instrument_platform':'ILLUMINA', 'single_end':true, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fastq/ERX5474936_ERR5766180_1.fastq.gz]]
[DUMP: diamond_test-noprocessing_reads] [['id':'2613', 'run_accession':'ERR5766181', 'instrument_platform':'ILLUMINA', 'single_end':false, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fastq/ERX5474937_ERR5766181_1.fastq.gz, /nf-core/test-datasets/taxprofiler/data/fastq/ERX5474937_ERR5766181_2.fastq.gz]]
[DUMP: diamond_test-noprocessing_reads] [['id':'ERR3201952', 'run_accession':'ERR3201952', 'instrument_platform':'OXFORD_NANOPORE', 'single_end':true, 'is_fasta':false, 'type':'long', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fastq/ERR3201952.fastq.gz]]
```

Reply:
@LilyAnderssonLee what does the resolvd `.command.sh` look like?

Reply:
Here is the error message and no `command.sh` file:
```
Caused by:
  No signature of method: nextflow.util.BlankSeparatedList.getExtension() is applicable for argument types: () values: [] -- Check script './workflows/../subworkflows/local/../../modules/nf-core/diamond/blastx/main.nf' at line: 33

Source block:
  def args = task.ext.args ?: ''
  def prefix = task.ext.prefix ?: "${meta.id}"
  def is_compressed = fasta.getExtension() == "gz" ? true : false
  def fasta_name = is_compressed ? fasta.getBaseName() : fasta
  def columns = blast_columns ? "${blast_columns}" : ''
  switch ( out_ext ) {
          case "blast": outfmt = 0; break
          case "xml": outfmt = 5; break
          case "txt": outfmt = 6; break
          case "daa": outfmt = 100; break
          case "sam": outfmt = 101; break
          case "tsv": outfmt = 102; break
          case "paf": outfmt = 103; break
          default:
              outfmt = '6';
              out_ext = 'txt';
              log.warn("Unknown output file format provided (${out_ext}): selecting DIAMOND default of tabular BLAST output (txt)");
              break
      }
  """
      if [ "${is_compressed}" == "true" ]; then
          gzip -c -d ${fasta} > ${fasta_name}
      fi
  
      DB=`find -L ./ -name "*.dmnd" | sed 's/\\.dmnd\$//'`
  
      diamond \\
          blastx \\
          --threads ${task.cpus} \\
          --db \$DB \\
          --query ${fasta_name} \\
          --outfmt ${outfmt} ${columns} \\
          ${args} \\
          --out ${prefix}.${out_ext} \\
          --log
  
      mv diamond.log ${prefix}.log
  
      cat <<-END_VERSIONS > versions.yml
      "${task.process}":
          diamond: \$(diamond --version 2>&1 | tail -n 1 | sed 's/^diamond version //')
      END_VERSIONS
      """

```